PWGHF/Tasks/taskMcValidation.cxx

```diff
@@ -344,8 +373,14 @@ struct HfTaskMcValidationGen {
         }
 
         if (nDaughters[iD] == 4) {
-          if (!RecoDecay::isMatchedMCGen<false, true>(mcParticles, particle, PDGArrayParticle[iD], arrPDGFinal4Prong[iD], true, nullptr, maxDepthForSearch[iD], &listDaughters)) {
-            continue;
+          if (iD != B0ToDminusPi) {
+            if (!RecoDecay::isMatchedMCGen<false, true>(mcParticles, particle, PDGArrayParticle[iD], arrPDGFinal4Prong[iD], true, nullptr, maxDepthForSearch[iD], &listDaughters)) {
+              continue;
+            }
+          } else { // For B0 we consider flavour oscillations
+            if (!RecoDecay::isMatchedMCGen<true, true>(mcParticles, particle, PDGArrayParticle[iD], arrPDGFinal4Prong[iD], true, nullptr, maxDepthForSearch[iD], &listDaughters)) {
+              continue;
```

Comment:
Here I would be consistent with the following check, so either `RecoDecay::isMatchedMCGen<true, true>` or `RecoDecay::isMatchedMCGen<true, false>` everywhere. The second `true` removes particles originating from interactions with the material when performing the matching, this should not have any effect for the B-meson decays.